modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/AssetInternalCategoryIdsKeywordQueryContributor.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.search.internal.spi.model.query.contributor;
+
+import com.liferay.portal.kernel.search.BooleanQuery;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.search.query.QueryHelper;
+import com.liferay.portal.search.spi.model.query.contributor.KeywordQueryContributor;
+import com.liferay.portal.search.spi.model.query.contributor.helper.KeywordQueryContributorHelper;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Rubén Pulido
+ */
+@Component(immediate = true, service = KeywordQueryContributor.class)
+public class AssetInternalCategoryIdsKeywordQueryContributor
+	implements KeywordQueryContributor {
+
+	@Override
+	public void contribute(
+		String keywords, BooleanQuery booleanQuery,
+		KeywordQueryContributorHelper keywordQueryContributorHelper) {
+
+		SearchContext searchContext =
+			keywordQueryContributorHelper.getSearchContext();
+
+		if (!searchContext.isIncludeInternalAssetCategories()) {
+			return;
+		}
+
+		queryHelper.addSearchTerm(
```

Comment:
Hey @ruben-pulido, 

I think this probably is not going to work, since searchTems come from keywords and attributes.

Usually assetCategoryIds are filled from  searchContext.getAssetCategoryIds();

Reply:
I am trying this approach:

`@Component(immediate = true, service = QueryPreFilterContributor.class)
public class AssetInternalCategoryIdsQueryPreFilterContributor
	implements QueryPreFilterContributor {

	@Override
	public void contribute(
		BooleanFilter fullQueryBooleanFilter, SearchContext searchContext) {

		long[] assetCategoryIds = searchContext.getAssetCategoryIds();

		if (ArrayUtil.isEmpty(assetCategoryIds)) {
			return;
		}

		if (!searchContext.isIncludeInternalAssetCategories()) {
			return;
		}

		TermsFilter termsFilter = new TermsFilter(
			Field.ASSET_INTERNAL_CATEGORY_IDS);

		termsFilter.addValues(ArrayUtil.toStringArray(assetCategoryIds));

		fullQueryBooleanFilter.add(termsFilter, BooleanClauseOccur.MUST);
	}

}`

But is not working ffor me :(